community/modules/scheduler/cloud-batch-job/main.tf

```diff
@@ -39,6 +40,50 @@ locals {
 
   subnetwork_name    = var.subnetwork != null ? var.subnetwork.name : "default"
   subnetwork_project = var.subnetwork != null ? var.subnetwork.project : var.project_id
+
+  # Filter network_storage for native Batch support
+  native_batch_fstype = ["nfs"]
+  native_batch_network_storage = [
+    for ns in var.network_storage :
+    ns if contains(local.native_batch_fstype, ns.fs_type)
+  ]
+  startup_script_network_storage = [
+    for ns in var.network_storage :
+    ns if !contains(local.native_batch_fstype, ns.fs_type)
+  ]
+
+  # Pull out runners to include in startup script
+  storage_client_install_runners = [
+    for ns in local.startup_script_network_storage :
+    ns.client_install_runner if ns.client_install_runner != null
+  ]
+  mount_runners = [
+    for ns in local.startup_script_network_storage :
+    ns.mount_runner if ns.mount_runner != null
+  ]
+
+  startup_script_runner = var.startup_script == null ? [] : [{
+    content     = var.startup_script
+    destination = "passed_startup_script.sh"
+    type        = "shell"
+  }]
+
+  full_runner_list = concat(
+    local.storage_client_install_runners,
+    local.mount_runners,
+    local.startup_script_runner
+  )
+}
+
+module "batch_job_startup_script" {
+  source = "github.com/GoogleCloudPlatform/hpc-toolkit//modules/scripts/startup-script?ref=v1.7.0"
+  count  = length(local.full_runner_list) > 0 ? 1 : 0
```

Comment:
It was discussed offline: This is the source of test failure.

@nick-stroud laid out a couple options (paraphrased, hopefully correctly):
1. Create the startup-script module regardless, even if not runners are present
2. Pass an addition "known at apply" flag from the batch module to the startup module

One other option I think could work is that we could update the startup-script module to recognize when no startup script runners are present, bypass creating the bucket and return an empty string as the startup script.

Reply:
As this PR is already getting large, I am opting for option 1. I will create a follow on task to handle this more elegantly, which I think possibly could be accomplished as you suggest, with startup script skipping the bucket creation.